### PR TITLE
don't fire notifications on initialization

### PR DESCRIPTION
**Pull request by bridiver**
fix https://github.com/brave/brave-browser/issues/9481

Resolves 

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by bsclifton** [comment]
I ran all tests (unit/browser) locally on Windows and everything passed 😄👍

File: components/content_settings/core/browser/brave_content_settings_pref_provider.h [hunk 1]
```diff
@@ -75,6 +75,8 @@ class BravePrefProvider : public PrefProvider,
   std::map<bool /* is_incognito */, std::vector<Rule>> cookie_rules_;
   std::map<bool /* is_incognito */, std::vector<Rule>> brave_cookie_rules_;
 
+  bool initialized_;
```
Comment (iefremov) [inline components/content_settings/core/browser/brave_content_settings_pref_provider.h:78]: we need a comment, explaining why this is needed

Comment (bridiver) [reply components/content_settings/core/browser/brave_content_settings_pref_provider.h:78]: add here https://github.com/brave/brave-core/pull/6126/files#diff-9981585c3d9184fc54b96697c59d2028R144

**Comment by kjozwiak** [comment]
Reproduced the original issue using the following two builds:

```
Brave | 1.11.97 Chromium: 84.0.4147.89 (Official Build) (64-bit)
-- | --
Revision | 19abfe7bcba9318a0b2a6bc6634a67fc834aa592-refs/branch-heads/4147@{#852}
OS | macOS Version 10.15.6 (Build 19G73)
```

```
Brave | 1.13.31 Chromium: 84.0.4147.89 (Official Build) nightly (64-bit)
-- | --
Revision | 19abfe7bcba9318a0b2a6bc6634a67fc834aa592-refs/branch-heads/4147@{#852}
OS | macOS Version 10.15.6 (Build 19G73)
```

* browser becomes none-responsive for ~10 seconds or so when opening the profile provided by @bsclifton.

Verification **PASSED** on `macOS 10.15.6` using the following build:

```
Brave | 1.13.36 Chromium: 84.0.4147.89 (Official Build) nightly (64-bit)
-- | --
Revision | 19abfe7bcba9318a0b2a6bc6634a67fc834aa592-refs/branch-heads/4147@{#852}
OS | macOS Version 10.15.6 (Build 19G73)
```

* ensured that the browser didn't freeze/become none-responsive when launching using the provided profile.

**Pull request merged**